and-cli-deploy-jenkins.js

```diff
@@ -0,0 +1,184 @@
+                echo.error("Please specify a profile [--profile] to trigger");
+                shell.exit(1);
+            }
+            const config = jenkins.getConfig();
+            if (config.profiles.indexof(profile) < 0) {
+                echo.error(
+                    "Profile not found, please check your configuration"
+                );
+                shell.exit(1);
+            }
+
+            job = config.profiles[profile];
+            tag = await userPrompt.questionAsync("Tag name to deploy: ");
+            this.validateOrExit();
+
+            await prompt.confirmOrExit(
+                `Please confirm deployment of tag ${tag} to ${job}`,
+                0
+            );
+
+            echo.message(`Jenkins starting ${job}, deploying version ${tag}`);
+            const url = `${config.url}/job/${job}/buildWithParameters?version=${tag}`;
+
+            const cmd = commandStringFactory.build(
+                "curl",
+                "-X",
+                "POST",
+                "-L",
+                `--user ${config.username}:${config.token}`,
+                url
+            );
+
+            if (shell.exec(cmd, { silent: false }).code !== 0) {
+                echo.error(" - Failed to deploy jenkins");
+                shell.exit(1);
+            }
+        },
+        validateOrExit() {
+            const errors = [];
+            const config = jenkins.getConfig();
+            if (
+                StringUtils.isEmpty(config.url) ||
+                StringUtils.isEmpty(config.username) ||
+                StringUtils.isEmpty(config.token)
+            ) {
+                errors.push("Credentials not found");
+            }
+            if (StringUtils.isEmpty(job)) {
+                errors.push("job is required");
+            }
+
+            if (StringUtils.isEmpty(tag)) {
+                errors.push("tag is required");
+            }
+
+            if (errors.length > 0) {
+                echo.errors(errors);
+                shell.exit(1);
+            }
+            return true;
+        },
+
+        async run() {
+            if (program.init) {
+                await this.init();
+            }
+            if (program.new) {
+                await this.createProfile();
+            }
+            await this.build();
+        },
+    };
+
+    // #endregion Functions
+
+    // -----------------------------------------------------------------------------------------
+    // #region Entrypoint
+    // -----------------------------------------------------------------------------------------
+
+    program
+        .description("Trigger a build through Jenkins remotely")
+        .option(CREATE_PROFILE_OPTION, "Profile selected to trigger")
+        .option(INIT_OPTION, "Configure Jenkins credentials")
+        .option(PROFILE_OPTION, "Profile selected to trigger")
+        .parse(process.argv);
+
+    // #endregion Entrypoint
+
+    await jenkinsDeploy.run();
```

Comment:
Might be worth adding in logic here before this `run()` call to display the help menu for users trying to peek at the available options. @wintondeshong added a helper in the `js` module, named `hasNoArguments` which does the program.args slicing for you. I think a recent example is the `and-cli-github.js` command:

```
    // If no options are passed in, output help and exit
    if (js.hasNoArguments()) {
        program.help();
    }
```